subprojects/docs/src/samples/providers/fileAndDirectoryProperty/build.gradle

```diff
@@ -0,0 +1,31 @@
+class FooExtension {
+    final DirectoryProperty someDirectory
+    final RegularFileProperty someFile
+    final ConfigurableFileCollection someFiles
+
+    FooExtension(Project project) {
+        someDirectory = project.layout.directoryProperty()
+        someFile = project.layout.fileProperty()
+        someFiles = project.files()
+    }
+}
+
+project.extensions.create('foo', FooExtension, project)
+
+foo {
+    someDirectory = project.layout.projectDirectory.dir('some-directory')
+    someFile = project.layout.buildDirectory.file('some-file')
+    someFiles.from project.files(someDirectory, someFile)
+}
+
+task print {
+    doLast {
+        def someDirectory = project.foo.someDirectory.get().asFile
+        logger.lifecycle("foo.someDirectory = " + someDirectory)
```

Comment:
Log level `quiet` is probably more appropriate.

Reply:
I converted every `lifecycle` to `quiet`.